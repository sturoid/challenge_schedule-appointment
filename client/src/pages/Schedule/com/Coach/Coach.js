import React from 'react';
import { format } from 'date-fns';
import { getScheduleTime } from '../../../../lib/lists/lists';
import setAvailability from './Coach.utils';
import { getDateFromTimeSlot } from '../../utils';
import './Coach.scss';

const CoachSchedule = ({ appointments = [], bookTimeSlot, coach }) => {
  const timeSlots = setAvailability(appointments, getScheduleTime());

  return (
    <div className="coach-time-slots">
      {timeSlots.map(ts => {
        return (
          <div
            key={String(ts.hours + ts.minutes)}
            className={`coach-time-slot${ts.booked ? ' booked' : ''}`}
            onClick={() => bookTimeSlot(ts, coach)}
            onKeyDown={() => bookTimeSlot(ts, coach)}
            role="button"
            tabIndex="0"
          >
            {format(getDateFromTimeSlot(ts), 'hh:mma')}
          </div>
        );
      })}
    </div>
  );
};

const Coach = ({ coach, bookTimeSlot }) => {
  return (
    <div className="coach">
      <div className="coach-name">{coach.name}</div>
      <CoachSchedule
        appointments={coach.appointments}
        coach={coach}
        bookTimeSlot={bookTimeSlot}
      />
    </div>
  );
};

export default Coach;
