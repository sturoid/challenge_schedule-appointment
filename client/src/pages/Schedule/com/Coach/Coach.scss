.coach {
  padding-bottom: 20px;
  .coach-name {
    padding: 20px;
    font-size: 2rem;
    font-weight: 300;
    color: var(--grey800);
  }

  .coach-time-slots {
    .coach-time-slot {
      border-bottom: 1px solid var(--grey200);
      color: var(--grey700);
      padding: 15px 0;
      display: flex;
      align-items: center;
      justify-content: center;
      transition: all 0.2s ease;
      &:hover:not(.booked) {
        background: var(--blueLight);
        color: var(--white);
        cursor: pointer;
      }
    }
    .booked {
      color: var(--grey500);
      background: var(--blue);
    }
  }
}
