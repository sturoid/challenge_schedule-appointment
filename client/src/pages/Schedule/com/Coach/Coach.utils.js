export default function setAvailability(appointments, timeSlots) {
  const tsWithBooked = [...timeSlots];

  appointments.forEach(a => {
    const start = new Date(Number(a.start));
    const index = tsWithBooked.findIndex(
      t => t.hours === start.getHours && t.minutes === start.getMinutes()
    );
    if (index !== -1) {
      tsWithBooked[index].booked = true;
    }
    tsWithBooked[0].booked = true;
  });

  return tsWithBooked;
}
