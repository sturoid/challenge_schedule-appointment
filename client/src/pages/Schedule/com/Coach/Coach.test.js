import React from 'react';
import renderer from 'react-test-renderer';
import Coach from './Coach';

const coach = {
  id: '1',
  name: 'Bob Test',
  appointments: [
    {
      id: '1',
      start: new Date('2019-12-11T11:30:00').toString(),
      end: new Date('2019-12-11T12:00:00').toString()
    },
    {
      id: '2',
      start: new Date('2019-12-11T14:00:00').toString(),
      end: new Date('2019-12-11T14:30:00').toString()
    }
  ]
};

describe('Coach', () => {
  it('renders correctly', () => {
    const com = renderer.create(<Coach coach={coach} />).toJSON();
    expect(com).toMatchSnapshot();
  });
});
