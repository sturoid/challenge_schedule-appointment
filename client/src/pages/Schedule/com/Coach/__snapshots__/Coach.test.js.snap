// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coach renders correctly 1`] = `
<div
  className="coach"
>
  <div
    className="coach-name"
  >
    Bob Test
  </div>
  <div
    className="coach-time-slots"
  >
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      12:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      12:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      01:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      01:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      02:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      02:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      03:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      03:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      04:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      04:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      05:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      05:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      06:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      06:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      07:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      07:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      08:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      08:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      09:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      09:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      10:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      10:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      11:00AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      11:30AM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      12:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      12:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      01:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      01:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      02:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      02:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      03:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      03:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      04:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      04:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      05:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      05:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      06:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      06:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      07:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      07:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      08:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      08:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      09:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      09:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      10:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      10:30PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      11:00PM
    </div>
    <div
      className="coach-time-slot"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      11:30PM
    </div>
  </div>
</div>
`;
